File: #0974.subarray-sums-divisible-by-k.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {

        unordered_map<int, int> m;
        m[0] = 1;
        int sum = 0;
        int result = 0;
        for (int num : nums) {
            sum = (sum + num) % k;
            if (sum < 0) {
                sum += k;
            }
            result += m[sum]++;
        }
        return result;
    }
};

File: #0974.subarray-sums-divisible-by-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraysDivByK = function(nums, k) {
    let total = 0;
    let count = 0;
    const seen = new Map([[0, 1]]);
    for (const num of nums) {
        total = ((total + num) % k + k) % k;
        const subCount = seen.get(total) ?? 0;
        count += subCount;
        seen.set(total, subCount + 1);
    }
    return count;
};
